### PR TITLE
Issue 298/task type config forms

### LINKED ISSUE #298: View and edit task config for task type
Task for epic #224 

**Story:** "A task manager can view and edit the settings that apply to a specific task type."
**Designs:** https://whimsical.com/zetkin-gen-3-wireframes-RQbRMqyhKBWSRzrxprGbAH

Open questions:
* Need to know the types of task configs in order to make the forms.
* Are the only initially supported task types: `offline`, `share_link`, `open_link`?
* If the type is `offline` do we need to show this card at all? How would this impact the layout.

### PR DESCRIPTION
**Pull request by djbusstop**
Resolves #298

### CODE REVIEW

File: src/components/organize/tasks/forms/utils.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import dayjs from 'dayjs';
+import { ZetkinTaskRequestBody } from 'types/tasks';
+
+export const isPublishedFirst = (values: ZetkinTaskRequestBody): boolean => {
```
Comment (richardolsson) [inline src/components/organize/tasks/forms/utils.ts:4]: So fresh and
so clean clean

File: src/components/organize/tasks/TaskTypeDetailsSection/index.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            ) }
+            { task.type === TASK_TYPE.SHARE_LINK && (
+                <ShareLinkDetails taskConfig={ task.config as ShareLinkConfig } />
+            ) }
+            { task.type === TASK_TYPE.COLLECT_DEMOGRAPHICS && (
+                <CollectDemographicsDetails taskConfig={ task.config as CollectDemographicsConfig } />
+            ) }
+
+            { /* Dialogs */ }
+            <ZetkinDialog
```
Comment (richardolsson) [inline src/components/organize/tasks/TaskTypeDetailsSection/index.tsx:55]: These aren't opened anywhere, are they?

**Review by richardolsson — COMMENTED** [review]
As always very clean and well structured. I did not find any issues with the code, except that I can't find anywhere to actually open the type-specific edit dialogs, in code or UI. Is that because it's still work in progress?

File: src/components/organize/tasks/forms/utils.ts (hunk 1 above)
Comment (djbusstop) [reply src/components/organize/tasks/forms/utils.ts:4]: i dunno lol i could have spent time making it cleaner but it does the job D:

File: src/components/organize/tasks/TaskTypeDetailsSection/index.tsx (hunk 2 above)
Comment (djbusstop) [reply src/components/organize/tasks/TaskTypeDetailsSection/index.tsx]: They have been removed

File: src/components/organize/tasks/forms/utils.ts (hunk 1 above)
Comment (richardolsson) [reply src/components/organize/tasks/forms/utils.ts:11]: Oh I was being serious though. I like the well-named functions that explain exactly what they're doing.

File: src/components/organize/tasks/forms/typeConfigFields/VisitLinkFields.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { TextField } from 'mui-rff';
+import { useIntl } from 'react-intl';
+
+import { VISIT_LINK_FIELDS } from '../constants';
+
+const VisitLinkFields = (): JSX.Element => {
+    const intl = useIntl();
+    return (
+        <TextField
```
Comment (richardolsson) [inline src/components/organize/tasks/forms/typeConfigFields/VisitLinkFields.tsx:9]: Maybe we could have some sort of URL validation on this? Lets file a separate low-priority issue for that.

**Review by richardolsson — APPROVED** [review]
All good! Green light to merge. 🏁

File: src/components/organize/tasks/forms/typeConfigFields/VisitLinkFields.tsx (hunk 3 above)
Comment (richardolsson) [reply src/components/organize/tasks/forms/typeConfigFields/VisitLinkFields.tsx:9]: Filed as #345

**Review by richardolsson — APPROVED** [review]
Looking good! 💯

**Pull request merged**